cmd/generate/config/rules/telegram.go

```diff
@@ -0,0 +1,43 @@
+package rules
+
+import (
+	"math/rand"
+	"regexp"
+	"strconv"
+
+	"github.com/zricethezav/gitleaks/v8/cmd/generate/secrets"
+	"github.com/zricethezav/gitleaks/v8/config"
+)
+
+func TelegramBotToken() *config.Rule {
+	// define rule
+	r := config.Rule{
+		Description: "Telegram Bot API Token",
+		RuleID:      "telegram-bot-api-token",
+		SecretGroup: 1,
+		Regex:       regexp.MustCompile(`(?i)(?:^|[^0-9])([0-9]{5,16}:A[a-zA-Z0-9_\-]{34})(?:$|[^a-zA-Z0-9_\-])`),
+		Keywords: []string{
+			"telegram",
+			"api",
+			"bot",
+			"token",
+			"url",
+		},
+	}
+
+	// validate
+	token := secrets.NewSecret(numeric(strconv.Itoa(rand.Intn(11)+5)) + ":A" + alphaNumericExtendedShort("34"))
```

Comment:
math/rand is using to generate bot_id length in the test token. Seems it isn't an issue.
#jit_ignore_finding

Reply:
Rather than using Intn(11), I'd recommend breaking this up into non-random cases to test the corner cases explicitly:

**True positives (tps)**
1. min length (5?)
2. max length (15?)

**False positives**
1. min length - 1 (4?)
2. max length + 1 (16?)

Reply:
@weineran sounds reasonable, thanks.
Done by commit #b245919